streams/src/components/rtsp/header.ts

```diff
@@ -0,0 +1,184 @@
+ * SWS             =  [LWS] ; Separating whitespace
+ * HCOLON          =  *( SP / HT ) ":" SWS
+ *
+ * RTSP response rules (a `*` means zero or more):
+ * ====
+ * Status-Line  = RTSP-Version SP Status-Code SP Reason-Phrase CRLF
+ * Response     = Status-Line
+ *                *((general-header
+ *                /  response-header
+ *                /  message-body-header) CRLF)
+ *                CRLF
+ *                [ message-body-data ]
+ *
+ * header specifications:
+ *
+ * Content-Base       =  "Content-Base" HCOLON RTSP-URI
+ * Content-Length     =  "Content-Length" HCOLON 1*19DIGIT
+ * Content-Location   =  "Content-Location" HCOLON RTSP-REQ-Ref
+ * Session            =  "Session" HCOLON session-id
+ *                       [ SEMI "timeout" EQUAL delta-seconds ]
+ * session-id         =  1*256( ALPHA / DIGIT / safe )
+ * delta-seconds      =  1*19DIGIT
+ * Connection         =  "Connection" HCOLON connection-token
+ *                       *(COMMA connection-token)
+ * connection-token   =  "close" / token
+ * CSeq               =  "CSeq" HCOLON cseq-nr
+ * cseq-nr            =  1*9DIGIT
+ * Range              =  "Range" HCOLON ranges-spec
+ * ranges-spec        =  npt-range / utc-range / smpte-range
+ *                       /  range-ext
+ * npt-range          =  "npt" [EQUAL npt-range-spec]
+ * npt-range-spec     =  ( npt-time "-" [ npt-time ] ) / ( "-" npt-time )
+ * npt-time           =  "now" / npt-sec / npt-hhmmss / npt-hhmmss-comp
+ * npt-sec            =  1*19DIGIT [ "." 1*9DIGIT ]
+ * npt-hhmmss         =  npt-hh ":" npt-mm ":" npt-ss [ "." 1*9DIGIT ]
+ * npt-hh             =  2*19DIGIT   ; any positive number
+ * npt-mm             =  2*2DIGIT  ; 0-59
+ * npt-ss             =  2*2DIGIT  ; 0-59
+ * npt-hhmmss-comp    =  npt-hh-comp ":" npt-mm-comp ":" npt-ss-comp
+ *                       [ "." 1*9DIGIT ] ; Compatibility format
+ * npt-hh-comp        =  1*19DIGIT   ; any positive number
+ * npt-mm-comp        =  1*2DIGIT  ; 0-59
+ * npt-ss-comp        =  1*2DIGIT  ; 0-59
+ *
+ * Example response:
+ * ====
+ * RTSP/1.0 200 OK
+ * CSeq: 3
+ * Content-Type: application/sdp
+ * Content-Base: rtsp://192.168.0.3/axis-media/media.amp/
+ * Server: GStreamer RTSP server
+ * Date: Wed, 03 Jun 2015 14:23:42 GMT
+ * Content-Length: 623
+ *
+ * v=0
+ * ....
+ */
+
+/**
+ * Parse headers from an RTSP response. The body can be included as the
+ * parser will stop at the first empty line.
+ *
+ * First line is "start-line":
+ * <RTSP-Version> SP <Status-Code> SP <Reason Phrase> CRLF
+ *
+ * Rest is actual headers:
+ * Content-Base       =  "Content-Base" HCOLON RTSP-URI
+ */
+export function parseResponse(response: string): {
+  statusCode: number
+  headers: Headers
+} {
+  const messageLines = response
+    .trimStart()
+    .split('\n')
+    .map((line) => line.trim())
+  const [startline, ...headerlines] = messageLines
+
+  const [_rtspVersion, statusCode, _reasonPhrase] = startline.split(' ')
```

Comment:
Why are you extracting variables you're not using?

Reply:
I thought it would be nice as documention-like placeholders.